garage/tf/algos/vpg.py

```diff
@@ -94,16 +93,30 @@ def init_opt(self):
 
             # formulate as a minimization problem
             # The gradient of the surrogate objective is the policy gradient
-            if is_recurrent:
-                surr_obj = -tf.reduce_sum(logli * advantage_var *
-                                          valid_var) / tf.reduce_sum(valid_var)
-                mean_kl = tf.reduce_sum(
-                    kl * valid_var) / tf.reduce_sum(valid_var)
-                max_kl = tf.reduce_max(kl * valid_var)
-            else:
-                surr_obj = -tf.reduce_mean(logli * advantage_var)
-                mean_kl = tf.reduce_mean(kl)
-                max_kl = tf.reduce_max(kl)
+            with tf.name_scope(
+                    "surr_obj", values=[logli, advantage_var, valid_var]):
+                if is_recurrent:
+                    surr_obj = (
+                        -tf.reduce_sum(logli * advantage_var * valid_var) /
+                        tf.reduce_sum(valid_var))
+                else:
+                    surr_obj = -tf.reduce_mean(logli * advantage_var)
+                tf.identity(surr_obj, name="surr_obj")
+
+            with tf.name_scope("mean_kl", values=[kl, valid_var]):
+                if is_recurrent:
+                    mean_kl = tf.reduce_sum(
+                        kl * valid_var) / tf.reduce_sum(valid_var)
+                else:
+                    mean_kl = tf.reduce_mean(kl)
+                tf.identity(mean_kl, name="mean_kl")
+
+            with tf.name_scope("max_kl", values=[kl, valid_var]):
+                if is_recurrent:
+                    max_kl = tf.reduce_max(kl * valid_var)
+                else:
+                    max_kl = tf.reduce_max(kl)
+                tf.identity(max_kl, name="max_kl")
```

Comment:
I like the way you separated the code here!

Reply:
Thanks!